Weekly view

noitcudni opened the pull request:
![image](https://cloud.githubusercontent.com/assets/78951/21552088/24a8f0e6-cdcd-11e6-92b0-aad3e9fcef5d.png)

Added a new property 'calendarFormat,' which can either be "monthly" or "weekly".

Simplified the eventMap data structure. Instead of having a map of maps, it's now just a map keyed by date (YYYY-MM-DD) string. 

The more interesting change is in getStartMoment(). 

Please let me know what you think and if you'd like me to clarify and/or clean up anything. 

Closes #28

christopherdro commented:
Awesome! Thanks for the PR. Will try and review this weekend.

christopherdro commented:
@noitcudni I am seeing a few issues with how the surrounding logic is works in this mode.

When I attempt to implement your changes on the example app of mater I see two issues that needs addressing before moving further.

- Dates are rendered incorrectly. 
- Month in header changes to the next month when it shouldn't.

![demo2](https://cloud.githubusercontent.com/assets/3741055/21966076/8696b052-db21-11e6-8485-0c63146b7d69.gif)

christopherdro commented on components/Calendar.js:
```diff
@@ -156,63 +155,81 @@ export default class Calendar extends Component {
+  getStartMoment(calFormat, currMoment) {
+    const weekStart = this.props.weekStart;
+
+    let r;
```
What is `r` signify here? Can we make this variable name more explicit?

noitcudni replied:
This is "borrowed" from the C convention. "r" is the return value, but I'll comply to you request.

christopherdro commented on components/Calendar.js:
```diff
@@ -94,44 +96,41 @@ export default class Calendar extends Component {
 
   onPrev = () => {
-    const newMoment = moment(this.state.currentMonthMoment).subtract(1, 'month');
+    const newMoment = this.props.calendarFormat == 'monthly' ?
```
strict equality `===`

christopherdro commented on components/Calendar.js:
```diff
@@ -94,44 +96,41 @@ export default class Calendar extends Component {
 
   onNext = () => {
-    const newMoment = moment(this.state.currentMonthMoment).add(1, 'month');
+    const newMoment = this.props.calendarFormat == 'monthly' ?
```
strict equality

christopherdro commented on components/Calendar.js:
```diff
@@ -156,63 +155,81 @@ export default class Calendar extends Component {
+    const weekStart = this.props.weekStart;
+
+    let r;
+    if (calFormat == "monthly") {
```
single quotes + strict equality

noitcudni commented:
Thanks for the feedback. I'll take a look. I didn't know the option of swiping left and right was available.

noitcudni commented:
@christopherdro 
Thanks again for you detailed screen shot (how did you make that by the way?)

The following is what I fixed.
1) Make `scrollEnded()` aware of the weekly mode.
2) Preinstantiate prior and next 2 weeks in `getMonthStack()`, which I renamed to `getStack()`.

christopherdro commented:
@noitcudni Thanks for updating the PR. I will give it another go in the next days.

The app to create gif screenshots is called Licecap.
http://www.cockos.com/licecap/

noitcudni commented:
I noticed that the calendar can actually be "scrolled" vertically, so I changed `automaticallyAdjustContentInsets={false}` inside the `ScrollView.` Swiping left and right still work, while scrolling vertically doesn't do anything any longer.

amol-c commented:
Would love to see this PR get in.

coderdave commented:
@noitcudni Can this be resolved with master?

noitcudni commented:
I'll do it in the next couple days.

On Sun, Feb 12, 2017 at 8:02 AM, coderdave <notifications@github.com> wrote:

> @noitcudni <https://github.com/noitcudni> Can this be resolved with
> master?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/christopherdro/react-native-calendar/pull/87#issuecomment-279227957>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAE0Z55IM0X-qdIfmKTrecJem-VND-0Uks5rby0AgaJpZM4LXxlI>
> .
>

coderdave commented:
Also, `this.state.currentMonthMoment` is undefined on line 285 in `Calendar.js`. Should it be `this.state.currentMoment` now?

christopherdro commented:
Looks like the PR renames those state variables to exclude the word moment. I think we can leave that out for now as it's likely to continuously create conflicts with other PR's that might get merged before this one.

noitcudni commented:
So, should I stay put and wait?

On Mon, Feb 13, 2017 at 8:35 PM, Christopher Dro <notifications@github.com>
wrote:

> Looks like the PR renames those state variables to exclude the word
> moment. I think we can leave that out for now as it's likely to
> continuously create conflicts with other PR's that might get merged before
> this one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/christopherdro/react-native-calendar/pull/87#issuecomment-279605937>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAE0Z5-r0dLAXQrz4F9qcZS5kVjYyV08ks5rcS8sgaJpZM4LXxlI>
> .
>

GU5TAF commented:
Would love to see this get merged, has there been any progress @noitcudni @christopherdro ?

noitcudni commented:
I have been busy at work. I might be able to start working on this again soon. @christopherdro, would you want me to get started on this soon? Or, do you still want me to hold off and wait for other pull requests to get pulled in first.

christopherdro commented:
@GU5TAF @noitcudni Sorry for the late response. Been quite busy myself the last couple weeks.

I will spend some time this weekend looking over everything.

christopherdro commented:
Would be nice to wait for this to get merged since it has a significant amount of changes.
https://github.com/christopherdro/react-native-calendar/pull/107

joshjhargreaves commented:
What needs to be done to get this merged :)?

noitcudni commented:
I might have some free time now to look into the merge conflicts. Are you guys still trying to prioritize other pull requests?

noitcudni commented:
@coderdave @christopherdro I just did a 3 way merge from your master into my fork's master.  Please review at your earliest convenience.

MarsWang42 commented:
Hey @noitcudni thanks for the work! But it seems like it's not working properly on ios if I set `scrollenabled`. I currently on RN 0.44.0.

noitcudni commented:
@Marswang92. What do you mean not working properly? Does it scroll but displays the wrong dates? Or, it doesn't scroll at all?

MarsWang42 commented:
So if I set `scrollEnabled`, the height of calendar will be set as the same as `monthly` and dates will not show up on the first render.
![b7zumxncsz](https://cloud.githubusercontent.com/assets/14301745/26285305/209d9f56-3e02-11e7-926a-facfc8b32a0b.gif)
`scrollEnabled` removed, everything works well.
![ksaqibzsqs](https://cloud.githubusercontent.com/assets/14301745/26285312/2dfaf3ba-3e02-11e7-98a2-d9d47474d058.gif)
@noitcudni Any clue?

MarsWang42 commented:
Hey, the rendering problem has been solved via setting `removeClippedSubviews` to be false, my bad. But still no clue about the height issue.

MarsWang42 commented:
@noitcudni found out the reason and made a PR https://github.com/christopherdro/react-native-calendar/pull/139/

Would you mind check?

noitcudni commented:
@Marswang92 sure thing.

jaycee425 commented:
should the documentation be updated for these changes?
